CO-2745: Migrate partner_communication to V11

Nooka10 opened the pull request.

ecino reviewed (changes requested):
Formatting issues:
```
partner_communication/models/communication_job.py:204:31: E126 continuation line over-indented for hanging indent
partner_communication/models/communication_job.py:208:80: E501 line too long (80 > 79 characters)
partner_communication/models/communication_job.py:211:27: E126 continuation line over-indented for hanging indent
partner_communication/models/communication_job.py:541:34: E126 continuation line over-indented for hanging indent
```

Nooka10 committed:
CO-2745: Migrate partner_communication to V11


CO-2745: Migrate partner_communication to V11

Multiple debugs
CO-2745: Migrate partner_communication to V11

Made requested changes

ecino reviewed (changes requested):
The branch has some conflicts with changes that were imported from version 10.0. You should be able to fix them either by rebasing onto branch 11.0 or by merging branch 10.0 into yours. You have also one flake8 left:
`partner_communication/models/communication_job.py:540:34: E126 continuation line over-indented for hanging indent`

The travis build seems also not able to install pycups, you can try to fix that by editing the file requirements.txt or .travis.yml if you need to install the package with apt.
